bootstrap-salt.sh

```diff
@@ -1798,24 +1798,29 @@ __function_defined() {
 #                 process is finished so the script doesn't exit on a locked proc.
 #----------------------------------------------------------------------------------------------------------------------
 __wait_for_apt(){
-    echodebug "Checking if apt process is currently running."
-
     # Timeout set at 15 minutes
     WAIT_TIMEOUT=900
 
-    while ps -C apt,apt-get,aptitude,dpkg >/dev/null; do
-        sleep 1
-        WAIT_TIMEOUT=$((WAIT_TIMEOUT - 1))
-
-        # If timeout reaches 0, abort.
-        if [ "$WAIT_TIMEOUT" -eq 0 ]; then
-            echoerror "Apt, apt-get, aptitude, or dpkg process is taking too long."
-            echoerror "Bootstrap script cannot proceed. Aborting."
-            return 1
-        fi
+    # Run our passed in apt command
+    "${@}"
+    APT_RETURN=$?
+
+    # If our exit code from apt is 100, then we're waiting on a lock
+    while [ $APT_RETURN -eq 100 ]; do
+      sleep 1
+      WAIT_TIMEOUT=$((WAIT_TIMEOUT - 1))
+
+      # If timeout reaches 0, abort.
+      if [ "$WAIT_TIMEOUT" -eq 0 ]; then
+          echoerror "Apt, apt-get, aptitude, or dpkg process is taking too long."
+          echoerror "Bootstrap script cannot proceed. Aborting."
+          return 1
+      else
+          # Try running apt again until our return code != 100
+	  "${@}"
+    	  APT_RETURN=$?
+      fi
     done
-
-    echodebug "No apt processes are currently running."
 }
```

Comment:
Looks like you need to return the apt exit code in the end. It is required for the caller to be able to determine if it was not lock-related failure.
```suggestion

    return $APT_RETURN
}
```